readers/timescale/messages.go

```diff
@@ -42,7 +42,7 @@ func (tr timescaleRepository) ReadAll(chanID string, rpm readers.PageMetadata) (
 
 	// If aggregation is provided, add time_bucket and aggregation to the query
 	if rpm.Aggregation != "" {
-		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000 AS time, %s(value) AS value FROM %s WHERE %s GROUP BY 1 ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
+		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000 AS time, publisher, protocol, subtopic, name, unit, %s(value) AS value FROM %s WHERE %s GROUP BY 1, publisher, protocol, subtopic, name, unit ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
```

Comment:
In PostgreSQL, the `GROUP BY 1` clause is used to refer to the first column in the SELECT list, which in this case it is the `time` column derived from `EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000`.

Do we need group remove `BY 1`,  i.e group `BY time` ?
I don't think we need.

Reply:
Okay I agree, In that case can we have like `GROUP BY time` instead of `1`, Precisely by column name ?